src/main/java/com/couchbase/lite/util/Utils.java

```diff
@@ -81,6 +84,14 @@ public static boolean isPermanentError(int code) {
         return (code >= 400 && code <= 405) || (code == 407) || (code >= 409 && code <= 499);
     }
 
+    public static boolean isTransientErrorWithSpecialCases(Response response) {
+        return isTransientErrorWithSpecialCases(response.code());
+    }
+
+    public static boolean isTransientErrorWithSpecialCases(int code) {
+        return code == 406 || code == 408 || isTransientError(code);
```

Comment:
If CouchDB could send 406 and 408, shouldn't CBL iOS checks that as well? If iOS/.NET should check it, could you please check a ticket on those platforms?

For the answer, IMO, I don't think we should have a special case method.